### ISSUE TITLE
Serialization issue with custom learning rate schedule

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#973]
When saving a classifier with a custom learning rate schedule, we hit an error `unmatched objects: [<tf.Variable 'learning_rate:0' shape=() dtype=float32, numpy=5e-05>]`. I believe this is due to our default compilation.

Step to reproduce:
- Create a classifier model.
- Compile with a custom learning rate schedule
- Save and load the model
- Error!

[Colab](https://colab.research.google.com/gist/mattdangerw/0e5856708913a9bde06882c9c447e0b5/kerasnlp-custom-learning-rate-serialization-bug.ipynb)

**Comment by mattdangerw** [comment]
I believe this is because of our default compilation. If our saved model has different optimizer variables then the ones we create when we compile be default, we hit an error due too a mismatch in variables.

Possibly a fix here would be to not do default compilation when restoring from a saved model (though I am not sure how best to detect this).

**Comment by chenmoneygithub** [comment]
I think your theory is correct, as the code below works:
```
bert_backbone = keras_nlp.models.BertBackbone.from_preset(
    "bert_tiny_en_uncased",
)
bert_backbone.compile(
    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
    optimizer=tf.keras.optimizers.Adam(learning_rate=MySchedule(0.001)),
    metrics=["accuracy"],
    jit_compile=True,
)
bert_backbone.save("bert_backbone")
restored_model = keras.models.load_model("bert_backbone")
```

I am actually a bit confused on why it tries to find a match in the checkpoint. I was assuming `model.optimizer` would have already been restored before the `model.compile` call, so `model.compile` just overwrites the optimizer. Users can nullify the optimizer before saving to surpass the error, since usually optimizer is useless with the Keras saving format. Users hoping to resume training is encouraged to save a checkpoint instead of the graph.

**Comment by maitry63** [comment]
Hi @mattdangerw,
I looked into this issue & find out with the latest version of `KerasHub`, the model now saves and loads successfully, correctly restoring both the architecture and weights.

I think the remaining warnings regarding `optimizer` and `compile-state`restoration are expected behaviors rather than a failure in loading. 

**Compile State**: `compile()` is not automatically restored because BertClassifier overrides compile() and does not implement the full compile serialization hooks (get_compile_config / compile_from_config).

**Optimizer Warning**: The message "skipping variable loading..." simply indicates a mismatch between the saved optimizer state and the one reconstructed at load time, causing Keras to safely skip it.

Ultimately, this is a known limitation regarding the restoration of the full training state for subclassed models with custom learning-rate schedules and custom compile logic, rather than a bug.

I have attached the updated [gist](https://colab.sandbox.google.com/gist/maitry63/6fa5d5edbac12c6e4d63f82e3a4c3fde/kerasnlp-custom-learning-rate-serialization-bug.ipynb#scrollTo=4rc9v0ag2i1M) for reference. I think we can safely close this issue.

Let me know , if this observation helped.
Thank you !
